src/layouts/EditHomepage.jsx

```diff
@@ -43,22 +43,59 @@ import {
 import { HomepageStartEditingImage } from "assets"
 import { DEFAULT_RETRY_MSG } from "utils"
 
-import { DragDropContextProvider } from "../contexts/DragDropContext"
+import { EditableContextProvider } from "../contexts/EditableContext"
 import { useDrag, onCreate, onDelete } from "../hooks/useDrag"
 
 import { CustomiseSectionsHeader, Editable } from "./components/Editable"
 import { AddSectionButton } from "./components/Editable/AddSectionButton"
 import { HeroBody } from "./components/Homepage/HeroBody"
+import { HeroDropdownSection } from "./components/Homepage/HeroDropdownSection"
+import { HeroHighlightSection } from "./components/Homepage/HeroHighlightSection"
 import { InfobarBody } from "./components/Homepage/InfobarBody"
 import { InfopicBody } from "./components/Homepage/InfopicBody"
 import { ResourcesBody } from "./components/Homepage/ResourcesBody"
 
 /* eslint-disable react/no-array-index-key */
 
-// Constants
-// ==========
 const RADIX_PARSE_INT = 10
 
+// NOTE: `scrollIntoView` does not work when called synchronously
+// to avoid this problem, we do a `setTimeout` to wrap it.
+// This calls it after 1ms, which allows it to work.
+const scrollTo = (ref) => {
+  setTimeout(() =>
+    ref.current.scrollIntoView({
+      behavior: "smooth",
+    })
+  )
+}
+
+const getHasError = (errorArray) =>
+  _.some(errorArray, (err) =>
+    _.some(err, (errorMessage) => errorMessage.length > 0)
+  )
+
+const getHasErrors = (errors) => {
```

Comment:
nit: is the get necessary, can just be hasErrors ?

Reply:
It is because we're extracting a derived value (`hasError`) from a base value. The `get` is there to hint that this is an action